src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,200 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        Node(Node<T> prev, T value, Node<T> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    private void linkLast(T value) {
+        Node<T> currentTail = tail;
+        Node<T> newTail = new Node<>(currentTail, value, null);
+        tail = newTail;
```

Comment:
let's move `tail = newTail;` after if-else block and then we can remove line `Node<T> currentTail = tail;`